### ISSUE TITLE
Why my model file size is so big (551MB)?

### ISSUE THREAD

**Issue by happyfrank** [jcjohnson/fast-neural-style#47]
When I use this command
`th train.lua   -h5_file out/file.h5   -style_image out/starry_night.jpg   -style_image_size 384   -content_weights 1.0   -style_weights 5.0   -checkpoint_name checkpoint   -gpu 0`
is there anything wrong? @jcjohnson

**Comment by mynameisvinn** [comment]
@happyfrank @jcjohnson as far as i can tell, model weights > 550MB is not uncommon. in fact, the popular vgg model contained 567mb of parameter weights. 

but you should work out the math.

assuming image dimensions of 227x227x3, receptive field = 11, stride = 4, no padding, and conv depth of 96, this layer output volume would be of size 55x55x96. this means there 290,400 nodes (55*55*96)  in the first convolutional layer, each with 11*11*3 = 363 weights and 1 bias. collectively, this adds up to 290400 * 364 = 105,705,600 parameters on the first layer alone!

**Comment by htoyryla** [comment]
@mynameisvinn , I think the question here is why he is getting a file more than 20 times as big as the normal 20M when using fast-neural-style training defaults. One factor is that the checkpoint files generated contain also other things than the model weights and biases. Still over 500M seems a lot.

BTW, your calculation assumes no parameter sharing for a conv layer. I was under assumption that the convolution nodes in a layer share weights, so that the layer works invariant to the position of the features, and the layer adapts easily to images of different sizes.

**Comment by 66wangxuewen99** [comment]
for me is around 100M, I also wonder where is wrong?

**Comment by happyfrank** [comment]
I already solved it. because I didn't install python dependencies into a virtual environment this when I configuration the environment.
``virtualenv .env                  # Create the virtual environment
source .env/bin/activate         # Activate the virtual environment
pip install -r requirements.txt  # Install Python dependencies
# Work for a while ...
deactivate                       # Exit the virtual environment

**Comment by 66wangxuewen99** [comment]
@happyfrank  it seems doesn't work

**Comment by htoyryla** [comment]
I don't think the env has anything to do with the model size generated by train.lua. I haven't used env for this, because I had the required packages already in my global python environment. Python is also only used to generate the hdf5 file, not in the training of the model. I could be, though, that the size of the images (in the hdf5 file) used for training could affect the model size, but I have not checked whether this is the case.

**Comment by 66wangxuewen99** [comment]
I am sure i m look at model file not dataset file.the model file with a t7 extension is ~100M (use the default arch option)really don't know why.

**Comment by Runescaped** [comment]
@66wangxuewen99 , what's the exact command that you're running? I created a model with all the default settings, except that I set batch_size to 2. My resulting t7 file was 60MB.

**Comment by htoyryla** [comment]
I too found I have a couple of model files around 130M. It seems I have trained them using
```
 arch c3s1-32,d64,d128,R128,d256,R256,d512,R512,R512,u256,R256,u128,R128,u64,u32,c3s1-3 
```
which explains the size (larger model architecture and thus more weights).

Clearing the extra info away from the file using
```
th scripts/convert_network.lua -input_t7 bigfile.t7 -output_t7 modelonly.t7
```
brings the size down to 68M.

**Comment by xpeng** [comment]
I have same problem now. last month i've trained many models using AWS g2 and all things going ok, the pretrained model size is normally around 18MB.
recently, i got a new HPC as i can do more training works, the new HPC OS is CentOS, and i install all things as before, and use same parameters to train models with same h5 files i used. but the result pretrained model size is more than 55MB and the convert_network.lua reduce the size not too much.

i upgraded the linux cudnn libraries from Nvidia and i don't think this is the reason of the problem, do i? 
i tried to checkout different version source code from repo(now and before Nov) and not solve  the problem. 
i use the command as below and i don't know why the different thing happens here:
`th train.lua -h5_file datasets.h5 -arch c9s1-16,d32,d64,R64,R64,R64,R64,R64,u32,u16,c9s1-3 -tv_strength 0.0001 -style_image_size 384 -num_iterations 50000 -content_weights 1.0 -style_weights 30.0 -lr_decay_every 40000 -checkpoint_name style057_v2 -style_image styles/style057.jpg -gpu 0`
